### PR TITLE
Misc. small libgit2 tests

### PR DESCRIPTION
**Pull request by kshyatt**
Add tests for:
* Printing `ConfigEntry`
* Various credential equality things (super weaksauce tests rn just to make sure these functions work)
* `head` when you pass it a string that's a package location

### CODE REVIEW

File: test/libgit2.jl [hunk 1]
```diff
@@ -246,6 +246,8 @@ mktempdir() do dir
                 else
                     error("Found unexpected entry: $name")
                 end
+                show_str = sprint(show, entry)
+                @test show_str == string("ConfigEntry(\"", name, "\", \"", value, "\")")
```
Comment (omus) [inline test/libgit2.jl]: Just a style issue but I find using string interpolation makes things easier to read: 
```julia
"ConfigEntry(\"$name\", \"$value\")"
```

File: test/libgit2.jl [hunk 2]
```diff
@@ -1375,11 +1377,17 @@ mktempdir() do dir
         @test LibGit2.checkused!(creds)
         @test creds.user == creds_user
         @test creds.pass == creds_pass
+        creds2 = LibGit2.UserPasswordCredentials(creds_user, creds_pass)
+        @test creds == creds2
+        LibGit2.reset!(creds)
+        @test !LibGit2.checkused!(creds)
```
Comment (omus) [inline test/libgit2.jl]: I think having the `reset!` and `checkused!` moved up to above. Do you want this here to ensure that equality works when the count fields don't match?

File: test/libgit2.jl [hunk 3]
```diff
@@ -1375,11 +1377,17 @@ mktempdir() do dir
...
         sshcreds = LibGit2.SSHCredentials(creds_user, creds_pass)
         @test sshcreds.user == creds_user
         @test sshcreds.pass == creds_pass
         @test isempty(sshcreds.prvkey)
         @test isempty(sshcreds.pubkey)
+        sshcreds2 = LibGit2.SSHCredentials(creds_user, creds_pass)
+        @test sshcreds == sshcreds2
```
Comment (omus) [inline test/libgit2.jl]: Up to you but I would probably do:
```julia
@test sshcreds == LibGit2.SSHCredentials(creds_user, creds_pass)
```

File: test/libgit2.jl (hunk 2 above)
Comment (kshyatt) [inline test/libgit2.jl]: Yeah that was my thought

Comment (omus) [inline test/libgit2.jl]: 👍

**Comment by kshyatt** [comment]
@omus do you want me to change the equality test for SSHCreds or is this good to go :)?

**Review by omus — APPROVED** [review]
We're good to go

**Comment by ararslan** [comment]
Sorry, forgot to submit a review. Retroactively LGTM.

**Pull request merged**